apps/sms/test/marionette/conversation_test.js

```diff
@@ -314,4 +312,98 @@ marionette('Conversation Panel Tests', function() {
       assert.equal(downloadedMessage.content.trim(), 'Attachment');
     });
   });
+
+  suite('Edit Mode', function() {
+    var thread, conversationView;
+    setup(function() {
+      thread = ThreadGenerator.generate({
+        numberOfMessages: 4
+      });
+      messagesApp.launch();
+      storage.setMessagesStorage([thread], ThreadGenerator.uniqueMessageId);
+      storage.setContactsStorage();
+      var inboxView = new InboxView(client);
+      conversationView = inboxView.goToConversation(thread.id);
+
+    });
+    
+    test('Enter and leave edit', function () {
+
+      conversationView.enterEditMode();
+      conversationView.messages.forEach(function(message) {
+        assert.isTrue(message.isInEditMode);
+      });
+
+      conversationView.exitEditMode();
+      conversationView.messages.forEach(function(message) {
+        assert.isFalse(message.isInEditMode);
+      });
+
+    });
+
+    test.only('select/deselect all', function () {
```

Comment:
Let's turn this into suite ```Toggle selection button behavior``` and split into several tests:
* Select all when nothing is selected, deselect all - basically this test;
* Select one or two message manually and then press ```Select all``` and then ```Deselect all```;
* Select all messages with the button, then unselect some manually and tap on select button once again.